gnoi/reset/server.go

```diff
@@ -54,7 +62,26 @@ func (s *Server) Start(ctx context.Context, req *pb.StartRequest) (*pb.StartResp
 		return &pb.StartResponse{Response: &pb.StartResponse_ResetError{ResetError: resetError}}, nil
 	}
 
-	// TODO: Reset the Target.
+	s.reset()
 
 	return &pb.StartResponse{Response: &pb.StartResponse_ResetSuccess{}}, nil
```

Comment:
It's not clear to me why they ever return an error here since they indicate failures in the StartResponse. We might consider in the future changing this. Just food for thought!

Reply:
I think its the way the grpc handlers are setup. Not sure if theres a way to easily change the signature of them

Reply:
Such that we can send gRPC specific errors as oposed to service specific errors. We are doing the latter.
https://godoc.org/google.golang.org/grpc/status#Error

Reply:
Ack fair enough. It's more a thing of if we never set error, then its pretty useless but if we are following a particular design pattern then ok.